datasafe-simple-adapter/datasafe-simple-adapter-spring/src/main/java/de/adorsys/datasafe/simple/adapter/spring/factory/SpringSimpleDatasafeServiceFactory.java

```diff
@@ -0,0 +1,43 @@
+package de.adorsys.datasafe.simple.adapter.spring.factory;
+
+import de.adorsys.datasafe.simple.adapter.api.SimpleDatasafeService;
+import de.adorsys.datasafe.simple.adapter.api.exceptions.SimpleAdapterException;
+import de.adorsys.datasafe.simple.adapter.api.types.AmazonS3DFSCredentials;
+import de.adorsys.datasafe.simple.adapter.api.types.DFSCredentials;
+import de.adorsys.datasafe.simple.adapter.api.types.FilesystemDFSCredentials;
+import de.adorsys.datasafe.simple.adapter.impl.SimpleDatasafeServiceImpl;
+
+public class SpringSimpleDatasafeServiceFactory {
+
+    public SpringSimpleDatasafeServiceFactory(DFSCredentials dfsCredentials) {
+        this.dfsCredentials = dfsCredentials;
+    }
+
+    private DFSCredentials dfsCredentials;
+
+    public SimpleDatasafeService getSimpleDataSafeServiceWithSubdir(String subdirBelowRoot) {
```

Comment:
Maybe providing dfsCredentials as bean - so it will be direct implementation of AmazonS3DFSCredentials or  FilesystemDFSCredentials could drop this `if`

Reply:
yes, done as suggested